ts/components/dialog/user-settings/pages/user-pro/ProSettingsPage.tsx

```diff
@@ -0,0 +1,1008 @@
+  const { refetch } = useProAccessDetails();
+
+  const buttons = useMemo(() => {
+    return [
+      {
+        label: { token: 'retry' },
+        dataTestId: 'pro-backend-error-retry-button',
+        onClick: refetch,
+        closeAfterClick: true,
+      },
+      {
+        label: { token: 'helpSupport' },
+        dataTestId: 'pro-backend-error-support-button',
+        onClick: () => {
+          showLinkVisitWarningDialog('https://getsession.org/pro-form', dispatch);
+        },
+        closeAfterClick: true,
+      },
+    ] satisfies Array<LocalizedPopupDialogButtonOptions>;
+  }, [dispatch, refetch]);
+
+  return buttons;
+}
+
+function ProNonProContinueButton({ returnToThisModalAction, centerAlign }: SectionProps) {
+  const dispatch = useDispatch();
+  const neverHadPro = useCurrentNeverHadPro();
+  const { isLoading, isError } = useProAccessDetails();
+
+  const backendErrorButtons = useBackendErrorDialogButtons();
+
+  const handleClick = useCallback(() => {
+    if (isError) {
+      dispatch(
+        updateLocalizedPopupDialog({
+          title: { token: 'proStatusError' },
+          description: { token: 'proStatusNetworkErrorDescription' },
+          overrideButtons: backendErrorButtons,
+        })
+      );
+    } else if (isLoading) {
+      dispatch(
+        updateLocalizedPopupDialog({
+          title: { token: 'proStatusLoading' },
+          description: { token: 'proStatusLoadingDescription' },
+        })
+      );
+    } else {
+      dispatch(
+        userSettingsModal({
+          userSettingsPage: 'proNonOriginating',
+          nonOriginatingVariant: 'upgrade',
+          overrideBackAction: returnToThisModalAction,
+          centerAlign,
+        })
+      );
+    }
+  }, [dispatch, isLoading, isError, backendErrorButtons, centerAlign, returnToThisModalAction]);
+
+  if (!neverHadPro) {
+    return null;
+  }
+
+  return (
+    <SessionButton
+      {...proButtonProps}
+      buttonColor={
+        isLoading || isError ? SessionButtonColor.Disabled : SessionButtonColor.PrimaryDark
+      }
+      onClick={handleClick}
+      dataTestId="pro-open-platform-website-button"
+    >
+      <Localizer token="theContinue" />
+    </SessionButton>
+  );
+}
+
+const StatsContainer = styled.div``;
+
+const StatsRowContainer = styled.div`
+  display: flex;
+  gap: var(--margins-sm);
+  padding: var(--margins-md);
+`;
+
+const StatsItemContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: var(--margins-sm);
+  align-items: center;
+  flex-direction: row;
+  width: 50%;
+`;
+
+const StatsLabel = styled.div<{ disabled?: boolean }>`
+  font-size: var(--font-size-md);
+  color: ${props => (props.disabled ? 'var(--disabled-color)' : 'var(--text-primary-color)')};
+  font-weight: 700;
+  cursor: default;
+`;
+
+const proBoxShadow = '0 4px 4px 0 rgba(0, 0, 0, 0.25)';
+
+function ProStats() {
+  const proLongerMessagesSent = Storage.get(SettingsKey.proLongerMessagesSent) || 3000;
```

Comment:
maybe remove that 3000?

Reply:
have removed, will look at having a way to set mock values for them as that would be useful